sequencer/src/api/sql.rs

```diff
@@ -17,6 +31,80 @@ impl SequencerDataSource for DataSource {
 
         Ok(cfg.connect(provider).await?)
     }
+
+    async fn store_state<S: MerklizedState<SeqTypes>>(
+        &mut self,
+        path: MerklePath<S::Entry, S::Key, S::T>,
+        traversal_path: Vec<usize>,
+        block_number: u64,
+    ) -> anyhow::Result<()> {
+        <DataSource as UpdateStateData<SeqTypes, S>>::insert_merkle_nodes(
+            self,
+            path,
+            traversal_path,
+            block_number,
+        )
+        .await
+        .context("failed to insert merkle nodes! ")
+    }
+}
+
+#[async_trait]
+impl<D: SequencerDataSource + Send + Sync> UpdateStateStorage<SeqTypes, D> for ValidatedState {
```

Comment:
This implementation has nothing specific to SQL so I think it's confusing to be in this file. Might make more sense where `SequencerDataSource` is defined